src/commands/index.rs

```diff
@@ -147,16 +148,27 @@ where
         .and_then(|index| {
             let results: Result<Vec<_>, RedisError> = index
                 .search(&query)?
-                .map(|key| {
+                .filter_map(|key| {
```

Comment:
Why `filter_map`? Do you really want to ignore errors and skip them?

Reply:
@gavrie 
It's not ignoring errors only None, in case of error we should return Some(Err)